source/adapters/hip/usm_p2p.cpp

```diff
@@ -9,25 +9,59 @@
 //===----------------------------------------------------------------------===//
 
 #include "common.hpp"
+#include "context.hpp"
 
-UR_APIEXPORT ur_result_t UR_APICALL
-urUsmP2PEnablePeerAccessExp(ur_device_handle_t, ur_device_handle_t) {
-  detail::ur::die(
-      "urUsmP2PEnablePeerAccessExp is not implemented for HIP adapter.");
-  return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+UR_APIEXPORT ur_result_t UR_APICALL urUsmP2PEnablePeerAccessExp(
+    ur_device_handle_t commandDevice, ur_device_handle_t peerDevice) {
+  ur_result_t result = UR_RESULT_SUCCESS;
+  try {
+    ScopedContext active(commandDevice);
+    UR_CHECK_ERROR(hipDeviceEnablePeerAccess(peerDevice->get(), 0));
+  } catch (ur_result_t err) {
+    result = err;
+  }
+  return result;
 }
 
-UR_APIEXPORT ur_result_t UR_APICALL
-urUsmP2PDisablePeerAccessExp(ur_device_handle_t, ur_device_handle_t) {
-  detail::ur::die(
-      "urUsmP2PDisablePeerAccessExp is not implemented for HIP adapter.");
-  return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+UR_APIEXPORT ur_result_t UR_APICALL urUsmP2PDisablePeerAccessExp(
+    ur_device_handle_t commandDevice, ur_device_handle_t peerDevice) {
+  ur_result_t result = UR_RESULT_SUCCESS;
```

Comment:
Can you use the pattern where instead of using this `result` var you just return `err` in the `catch` and otherwise return `UR_RESULT_SUCCESS`?

Reply:
OK